### PR TITLE
Added ability to delete discount policy

### PR DESCRIPTION
**Pull request by shreyas-satish**

### CODE REVIEW

File: boxoffice/views/admin_discount.py [hunk 1]
```diff
@@ -152,13 +154,40 @@ def admin_edit_discount_policy(discount_policy):
...
+            errors=form.errors)
+
+    db.session.delete(discount_policy)
+    db.session.commit()
+    return api_success(result={}, doc="Discount policy deleted.", status_code=200)
+
+
+@app.route('/admin/o/<org_name>/discount_policy/<discount_policy_id>/delete', methods=['GET', 'POST'])
+@lastuser.requires_login
+@render_with({'text/html': 'index.html.jinja2', 'application/json': jsonify_delete_discount_policy})
```
Comment (jace) [inline boxoffice/views/admin_discount.py]: Is this really meant to render with `index.html.jinja2`?

File: boxoffice/views/admin_discount.py [hunk 2]
```diff
@@ -152,13 +154,40 @@ def admin_edit_discount_policy(discount_policy):
             errors=discount_policy_form.errors)
 
 
+def jsonify_delete_discount_policy(discount_policy_dict):
```
Comment (jace) [inline boxoffice/views/admin_discount.py]: This function is view logic. It is not a renderer. It should be a part of the view function.

File: boxoffice/views/admin_discount.py (hunk 1 above)
Comment (shreyas-satish) [reply boxoffice/views/admin_discount.py]: Yes, that's the boot up page for the SPA.

File: boxoffice/views/admin_discount.py (hunk 2 above)
Comment (shreyas-satish) [reply boxoffice/views/admin_discount.py]: Will renaming the function to `delete_discount_policy` work? It's been extracted out to make it work with `render_with`.

File: boxoffice/views/admin_discount.py (hunk 1 above)
Comment (jace) [reply boxoffice/views/admin_discount.py]: A comment to that effect will help.

File: boxoffice/views/admin_discount.py (hunk 2 above)
Comment (jace) [reply boxoffice/views/admin_discount.py]: `render_with` is the wrong tool here. Flask will let you specify separate handlers by content type, so use that and make separate views.

**Pull request merged**